src/test/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/AddressLinesMapperTest.java

```diff
@@ -0,0 +1,132 @@
+package uk.gov.hmcts.reform.civil.service.robotics.mapper;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.civil.model.Address;
+import uk.gov.hmcts.reform.civil.sampledata.AddressBuilder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class AddressLinesMapperTest {
+
+    AddressLinesMapper mapper = new AddressLinesMapper();
+
+    @Nested
+    class ShouldSplitAddressLines {
+
+        @Test
+        void shouldSplitAddressLine1_whenExceedsLimitAndContainsComma() {
+            Address address = Address.builder()
+                .addressLine1("1234567890, 12345678901234567890, abcdefghijk")
+                .build();
+
+            Address result = mapper.splitLongerLines(address);
+
+            assertThat(result).extracting("addressLine1")
+                .isEqualTo("1234567890, 12345678901234567890");
+            assertThat(result).extracting("addressLine2")
+                .isEqualTo("abcdefghijk");
+            assertThat(result).extracting("addressLine3")
+                .isNull();
+        }
+
+        @Test
+        void shouldSplitAddressLine1_whenExceedsLimitAndContainsCommaAsLastChar() {
+            Address address = Address.builder()
+                .addressLine1("12345678901234567890,1234567890abcdefghij,")
```

Comment:
Maybe we could also have a test when the 35th character is a blank/space following a comma?

Reply:
@swalker125  When after splitting individual line exceeds the limit of 35 then we return the original address as per A/C of the story. And here is test for that: https://github.com/hmcts/civil-service/pull/223/files#diff-9fa657a32e3076db810afcd08940e4b38a2bc7dde92be33017df5df75e1567b8R101